src/main/java/HtmlUtil.java

```diff
@@ -6,45 +6,37 @@
 
     public static String testableHtml(PageData pageData, boolean includeSuiteSetup) throws Exception {
         WikiPage wikiPage = pageData.getWikiPage();
-        StringBuffer buffer = new StringBuffer();
+        StringBuilder buffer = new StringBuilder();
 
         if (pageData.hasAttribute("Test")) {
             if (includeSuiteSetup) {
                 WikiPage suiteSetup = PageCrawlerImpl.getInheritedPage(SuiteResponder.SUITE_SETUP_NAME, wikiPage);
-                if (suiteSetup != null) {
-                    WikiPagePath pagePath = wikiPage.getPageCrawler().getFullPath(suiteSetup);
-                    String pagePathName = PathParser.render(pagePath);
-                    buffer.append("!include -setup .").append(pagePathName).append("\n");
-                }
+                extendBuffer(suiteSetup, buffer, "-setup");
             }
             WikiPage setup = PageCrawlerImpl.getInheritedPage("SetUp", wikiPage);
-            if (setup != null) {
-                WikiPagePath setupPath = wikiPage.getPageCrawler().getFullPath(setup);
-                String setupPathName = PathParser.render(setupPath);
-                buffer.append("!include -setup .").append(setupPathName).append("\n");
-            }
+            extendBuffer(setup, buffer, "-setup");
         }
 
         buffer.append(pageData.getContent());
         if (pageData.hasAttribute("Test")) {
             WikiPage teardown = PageCrawlerImpl.getInheritedPage("TearDown", wikiPage);
-            if (teardown != null) {
-                WikiPagePath tearDownPath = wikiPage.getPageCrawler().getFullPath(teardown);
-                String tearDownPathName = PathParser.render(tearDownPath);
-                buffer.append("!include -teardown .").append(tearDownPathName).append("\n");
-            }
+            extendBuffer(teardown, buffer, "-teardown");
             if (includeSuiteSetup) {
                 WikiPage suiteTeardown = PageCrawlerImpl.getInheritedPage(SuiteResponder.SUITE_TEARDOWN_NAME, wikiPage);
-                if (suiteTeardown != null) {
-                    WikiPagePath pagePath = wikiPage.getPageCrawler().getFullPath(suiteTeardown);
-                    String pagePathName = PathParser.render(pagePath);
-                    buffer.append("!include -teardown .").append(pagePathName).append("\n");
-                }
+                extendBuffer(suiteTeardown, buffer, "-teardown");
             }
         }
 
         pageData.setContent(buffer.toString());
         return pageData.getHtml();
     }
 
+    public static void extendBuffer(WikiPage wikiPage, StringBuilder buffer, String flag) throws Exception {
+        if (wikiPage != null) {
+            WikiPagePath pagePath = wikiPage.getPageCrawler().getFullPath(wikiPage);
+            String pagePathName = PathParser.render(pagePath);
+            buffer.append("!include ").append(flag).append(" .").append(pagePathName).append("\n");
```

Comment:
Maybe replace this line by formatting a string and return a string instead of adding the line to the buffer

Reply:
return empty string in case of null